### PR TITLE
Rename `classes` -> `plasma`

### LINKED ISSUE #744: Rename plasmapy.classes to plasmapy.plasma [PLEP7]
As part of the changes associated with the forthcoming PLEP 7 (PlasmaPy/PlasmaPy-PLEPs#26), we decided to rename ``plasmapy.classes`` to ``plasmapy.plasma``.  This is in an attempt to make the top-level namespace more structured and explicit.

This issue will breaking existing API.

### PR DESCRIPTION
**Pull request by StanczakDominik**
Closes #744

- [x] I have added a changelog entry for this pull request (please see
  `changelog/README.rst` for instructions, and if you need help with picking the PR type, ask!)
- [x] If adding new functionality, I have added (passing) tests and
      docstrings. (Tests pop up at the bottom, in the checks box).
- [x] I have fixed any new failing tests (if you're unsure why
  they're failing, ask!).

### CODE REVIEW

**Comment by StanczakDominik** [comment]
I think coveragepy just gave up on the rename, thus the coverage failure.

File: plasmapy/__init__.py [hunk 1]
```diff
@@ -35,11 +35,17 @@
...
+# Enforce Python version check during package import.
+# This is the same check as the one at the top of setup.py
+import sys
+
 # Packages may add whatever they like to this file, but
 # should keep this content at the top.
 # ----------------------------------------------------------------------------
 import pkg_resources
 
+from . import formulary, particles
```
Comment (rocco8773) [inline plasmapy/__init__.py:47]: maybe put these above with the `import sys`

File: plasmapy/plasma/sources/openpmd_hdf5.py [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
-from plasmapy.classes import GenericPlasma
-from plasmapy.classes.exceptions import DataStandardError
-
 import os
 from distutils.version import StrictVersion
 
+import astropy.units as u
+import numpy as np
+from plasmapy.plasma.exceptions import DataStandardError
+from plasmapy.plasma.plasma_base import GenericPlasma
```
Comment (rocco8773) [inline plasmapy/plasma/sources/openpmd_hdf5.py:7]:
```suggestion
import astropy.units as u
import numpy as np
import os

from distutils.version import StrictVersion

from plasmapy.plasma.exceptions import DataStandardError
from plasmapy.plasma.plasma_base import GenericPlasma
```

just my preference

File: plasmapy/plasma/sources/plasma3d.py [hunk 3]
```diff
@@ -2,15 +2,13 @@
...
-
+import astropy.units as u
+import numpy as np
 from astropy.constants import mu0
-
-from plasmapy.classes import GenericPlasma
+from plasmapy.formulary.magnetostatics import MagnetoStatics
+from plasmapy.plasma.plasma_base import GenericPlasma
 
 __all__ = ["Plasma3D"]
```
Comment (rocco8773) [inline plasmapy/plasma/sources/plasma3d.py:13]:
```suggestion
__all__ = ["Plasma3D"]

import astropy.units as u
import itertools
import numpy as np

from astropy.constants import mu0

from plasmapy.formulary.magnetostatics import MagnetoStatics
from plasmapy.plasma.plasma_base import GenericPlasma
```

PEP8: move `__all__` to the top
my preference for organizing imports

File: plasmapy/plasma/sources/plasmablob.py [hunk 4]
```diff
@@ -4,16 +4,13 @@
...
+from plasmapy.formulary.dimensionless import quantum_theta
+from plasmapy.formulary.parameters import _grab_charge
 from plasmapy.particles import particle_mass
-
+from plasmapy.plasma.plasma_base import GenericPlasma
 from plasmapy.utils import CouplingWarning, call_string
 
-from plasmapy.classes import GenericPlasma
-
 __all__ = ["PlasmaBlob"]
```
Comment (rocco8773) [inline plasmapy/plasma/sources/plasmablob.py:14]:
```suggestion
__all__ = ["PlasmaBlob"]

import astropy.units as u
import warnings

from plasmapy.formulary.collisions import coupling_parameter
from plasmapy.formulary.dimensionless import quantum_theta
from plasmapy.formulary.parameters import _grab_charge
from plasmapy.particles import particle_mass
from plasmapy.plasma.plasma_base import GenericPlasma
from plasmapy.utils import CouplingWarning, call_string
```

File: plasmapy/plasma/sources/tests/test_openpmd_hdf5.py [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
-from typing import Union, Tuple, List
 import os
+from typing import List, Tuple, Union
+
+import plasmapy.plasma
 import pytest
+from astropy import units as u
+from plasmapy.particles.data.test import data_dir
+from plasmapy.plasma.exceptions import DataStandardError
+from plasmapy.plasma.sources import openpmd_hdf5
```
Comment (rocco8773) [inline plasmapy/plasma/sources/tests/test_openpmd_hdf5.py:9]:
```suggestion
import astropy.units as u
import os
import pytest

from typing import List, Tuple, Union

import plasmapy.plasma

from plasmapy.particles.data.test import data_dir
from plasmapy.plasma.exceptions import DataStandardError
from plasmapy.plasma.sources import openpmd_hdf5
```

File: plasmapy/plasma/sources/tests/test_plasmablob.py [hunk 6]
```diff
@@ -1,10 +1,9 @@
-import pytest
-import numpy as np
 import astropy.units as u
-
-from plasmapy.classes.sources import plasma3d, plasmablob
+import numpy as np
+import pytest
```
Comment (rocco8773) [inline plasmapy/plasma/sources/tests/test_plasmablob.py:3]:
```suggestion
import pytest

```

File: plasmapy/plasma/sources/tests/test_plasma3d.py [hunk 7]
```diff
@@ -1,9 +1,8 @@
-import pytest
-import numpy as np
 import astropy.units as u
-
-from plasmapy.classes.sources import plasma3d
+import numpy as np
+import pytest
```
Comment (rocco8773) [inline plasmapy/plasma/sources/tests/test_plasma3d.py:3]:
```suggestion
import pytest

```

File: plasmapy/plasma/tests/test_plasma_factory.py [hunk 8]
```diff
@@ -1,41 +1,41 @@
+import os
+
 import astropy.units as u
 import numpy as np
-import os
+import plasmapy.plasma
 import pytest
-
-import plasmapy.classes
 from plasmapy.particles.data.test import data_dir
```
Comment (rocco8773) [inline plasmapy/plasma/tests/test_plasma_factory.py:7]:
```suggestion
import astropy.units as u
import numpy as np
import os
import pytest

import plasmapy.plasma

from plasmapy.particles.data.test import data_dir
```

File: plasmapy/simulation/tests/test_particletracker.py [hunk 9]
```diff
@@ -1,11 +1,10 @@
 import numpy as np
 import pytest
 from astropy import units as u
```
Comment (rocco8773) [inline plasmapy/simulation/tests/test_particletracker.py:3]:
```suggestion
import astropy.units as u
import numpy as np
import pytest

```

File: plasmapy/simulation/tests/test_particletracker.py [hunk 10]
```diff
@@ -1,11 +1,10 @@
...
-from astropy.modeling import models, fitting
-from scipy.optimize import curve_fit
-
+from astropy.modeling import fitting, models
+from plasmapy.plasma.sources import Plasma3D
 from plasmapy.simulation.particletracker import ParticleTracker
-from plasmapy.classes.sources import Plasma3D
+from scipy.optimize import curve_fit
```
Comment (rocco8773) [inline plasmapy/simulation/tests/test_particletracker.py:7]:
```suggestion
from astropy.modeling import fitting, models
from scipy.optimize import curve_fit

from plasmapy.plasma.sources import Plasma3D
from plasmapy.simulation.particletracker import ParticleTracker
```

**Comment by StanczakDominik** [comment]
I forgot to skip `black` checks, darn...

**Comment by StanczakDominik** [comment]
The changes from master I've just merged in should address most of the formatting issues, and the stability matrix :)

**Comment by StanczakDominik** [comment]
Thanks for the reviews @namurphy @rocco8773 !

**Pull request merged**